### PR TITLE
Clear info on flatpak wrapper, including feature differences from original app

### PR DESCRIPTION
**Pull request by geekley**
Please check to make sure I didn't write something stupid or incorrect, specially the legal section.

By the way, I think the owners should also add a LICENSE on the repo.

### CODE REVIEW

**Comment by geekley** [comment]
#99

File: README.md
Comment (lionirdeadman) [inline README.md:15]: This feels inappropriate to put in a page which is supposed to represent the application.

File: README.md
Comment (lionirdeadman) [inline README.md:21]: I would mention here that the plan is for Zypak to give us a file picker portal which will allow full access to the filesystem while still restricting unauthorized access.

File: com.discordapp.Discord.appdata.xml
Comment (lionirdeadman) [inline com.discordapp.Discord.appdata.xml:23]: I would say that it prevents "the following features from working out of the box" instead because otherwise it might imply that this can't be fixed by people using the application.

**Review by lionirdeadman — CHANGES_REQUESTED** [review]
This looks generally good but some twaaks should be made.

> By the way, I think the owners should also add a LICENSE on the repo.

I would normally agree but it seems that no Flathub repo has a license. I think that if we want to license these repositories, it should be fixed upstream to make sure everyone has the same license.

File: README.md
Comment (geekley) [reply README.md:21]: What would be the flatpak override command to allow access to Home? Is it `flatpak override --filesystem=home com.discordapp.Discord` ? Should I add this command as an alternative to Flatseal? @lionirdeadman

Comment (geekley) [reply README.md:21]:
```suggestion
- **Unrestricted File Access**: Default sandbox permissions limit Discord to only certain directories, so you can't access your entire Home directory. Currently, this limits which file directories you can attach files from and impacts drag and drop functionality.  
  This limitation will likely be overcomed eventually, when Zypak give us a file picker portal which will allow full access to the filesystem while still restricting unauthorized access.  
  To work around this now, you can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --filesystem=home com.discordapp.Discord`) to give Discord broader file system access, allowing file attachments from more locations.
```

File: com.discordapp.Discord.appdata.xml
Comment (geekley) [reply com.discordapp.Discord.appdata.xml:23]:
```suggestion
    <p>The flatpak version runs in a sandbox to provide better safety and privacy for users. However, this sandboxing prevents the following features from working out of the box: Game Activity, Unrestricted File Access, Rich Presence. Check the README in the Github repo for details.</p>
```

File: README.md
Comment (geekley) [reply README.md:15]:
```suggestion
The flatpak version runs in a sandbox to provide better safety and privacy for users.
```

File: README.md
Comment (lionirdeadman) [reply README.md:21]: It needs --user if it's installed as user application rather than system.

Comment (geekley) [reply README.md]: Well it's just an example anyways. It might also need sudo I think, if it's installed on system, but you'd get an error and know what to do. Feel free to rephrase it if needed.

Comment (geekley) [reply README.md]: Oh, okay. I didn't know you could add --user overrides on system installed apps too. I'll change it then.

File: README.md
Comment (geekley) [inline README.md:23]:
```suggestion
  To work around this now, you can change sandbox permissions of installed flatpak applications (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --user --filesystem=home com.discordapp.Discord`) to give Discord broader file system access, allowing file attachments from more locations.
```

File: README.md
Comment (lionirdeadman) [reply README.md]: You can't add --user to system installed apps. System and user applications have different permission overrides. Confusing, I know.

Comment (geekley) [reply README.md]: Are you sure? I just tested with a system installed Discord flatpak (confirmed with `flatpak info com.discordapp.Discord`). I did `flatpak override --user ...`, `flatpak info --show-permissions ...` and then again with `su otheruser` and it does show the overridden home permission for just one user.

Comment (geekley) [reply README.md]: Though maybe we should change it to `home:ro` if the purpose is only add attachments.
I didn't test this in the actual app by the way, whether just adding the `home` or `home:ro` permission actually allows file attachments properly. I'm assuming this works by what you guys have discussed.

Comment (lionirdeadman) [reply README.md]: Your flatpak info is for the system one, it also needs to be specificed --user.

`home:ro` would definitely work but I think realistically that zypak should get out of beta probably soon so it's not a mess I'm ready to make.

Comment (geekley) [reply README.md]: That's the thing, there's no user installation (`flatpak info --user ...` fails, with and without --show-permissions), I only have the default system installation for Discord.
Yet the --show-permissions shows the home permission only on the user that added it (with override --user). I haven't tested it, but this makes it seem that you **can** override it per user even on system wide installation. Either this is documented confusingly [here](https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpak-override), or maybe it's some kind of bug...?

**Comment by lionirdeadman** [comment]
With #132, the file picking capabilities will be resolved and instead only DND will not work, this should be reflected. Sorry for this~

File: README.md
Comment (geekley) [inline README.md:23]:
```suggestion
- **Drag and Drop Files**: To work around this now, you can change sandbox permissions of installed flatpak applications to give Discord file system access (for example, with [Flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) or with `flatpak override --user --filesystem=home:ro com.discordapp.Discord`).
```

File: com.discordapp.Discord.appdata.xml
Comment (geekley) [inline com.discordapp.Discord.appdata.xml:23]:
```suggestion
    </ul>
    <p>Discord is written in Javascript, React, Elixir, and Rust. The desktop application uses the electron framework.</p>
    <p>The flatpak version runs in a sandbox to provide better safety and privacy for users. However, this sandboxing prevents the following features from working out of the box: Game Activity, Drag and Drop Files, Rich Presence. Check the README in the Github repo for details.</p>
```

File: README.md
Comment (lionirdeadman) [reply README.md:23]: This seems good to me. Thanks for being fast and doing this btw! I'll merge this when #132 gets merged then.

File: com.discordapp.Discord.appdata.xml
Comment (TingPing) [inline com.discordapp.Discord.appdata.xml:23]: You can put a real link here for the readme.

Comment (geekley) [reply com.discordapp.Discord.appdata.xml:23]: Is it HMTL, like <a href... ?

File: README.md
Comment (TheEvilSkeleton) [inline README.md:21]:
```suggestion
- **Drag and Drop Files**: To work around this now, you can change sandbox permissions of installed flatpak applications to give Discord file system access:

```sh
flatpak override --filesystem=home:ro com.discordapp.Discord```
```

---

At line `8` you typed `flatpak install flathub com.discordapp.Discord` (without the `--user` flag), so when needing to change the permissions, you can't use `--user` since you installed it in your system.

IMO while I'm a big fan of Flatseal, I don't think instructions with Flatseal should be included in READMEs, or at least make it look better by doing a step-by-step guide rather than tell the user they can use Flatseal with no further instructions. It clutters a lot and makes it hard to follow through the README.

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]

File: README.md
Comment (geekley) [reply README.md:21]: > you can't use --user since you installed it in your system.

@TheEvilSkeleton I just tested this on my system installation of discord flatpak and yes it's working with --user override. See my comments above. https://github.com/flathub/com.discordapp.Discord/pull/133#discussion_r552984244

Comment (geekley) [reply README.md:21]: Without --user, I need sudo to override

Comment (TheEvilSkeleton) [reply README.md:21]: Wtf? I just tested it in a VM and it worked. Forget what I've said :P

Comment (geekley) [reply README.md:21]: flatpak docs is definetly confusing

Comment (TheEvilSkeleton) [reply README.md:21]: I updated the suggestion so it should look better.

Comment (TheEvilSkeleton) [reply README.md:21]: > flatpak docs is definetly confusing

Agreed

Comment (geekley) [reply README.md:21]: "allow edits by mantainers" is on, so you guys can commit whatever you want too, I think

Comment (TheEvilSkeleton) [reply README.md:21]: I'm not a maintainer, just a user lurking around.

Comment (geekley) [reply README.md:21]: Ah, ok. I'm a noob doing this, so I don't know how this all works

Comment (TheEvilSkeleton) [reply README.md:21]: Pretty sure you can press on "Accept changes", or something like that right?

Comment (geekley) [reply README.md:21]: yeah, I'll add that with the other link thing
Do you know if I can just add <a href=... on the appdata.xml?

Comment (TheEvilSkeleton) [reply README.md:21]: No idea.

File: com.discordapp.Discord.appdata.xml
Comment (geekley) [inline com.discordapp.Discord.appdata.xml:23]:
```suggestion
    <p>The flatpak version runs in a sandbox to provide better safety and privacy for users. However, this sandboxing prevents the following features from working out of the box: Game Activity, Drag and Drop Files, Rich Presence. Check <a href="https://github.com/flathub/com.discordapp.Discord#readme">the README</a> in the Github repo for details.</p>
```
@TingPing will this work?

Comment (TingPing) [reply com.discordapp.Discord.appdata.xml:23]: Reading the spec it doesn't actually support URLs, sorry for the misunderstanding.

**Comment by lionirdeadman** [comment]
Bad maintainer here with some bad news. Revert the change regarding zypak's ability to allow file picking ability. After testing, it turns out that it doesn't cover the full range of ways that Electron can pick files.

It'd also be nice if you made it one commit instead of 7 commits for the sake of history readability although I can just squash it instead if you want. Sorry for the hassle ^^"

**Comment by geekley** [comment]
@lionirdeadman Can you write on the repo? I have "allow edits by maintainers" on.
You can do this better than me, so if you have access, feel free to commit or squash anything on my branch.

**Comment by geekley** [comment]
@lionirdeadman There's also a "convert to draft" option. I don't know what it does, but tell me if you want me to use it.

**Comment by lionirdeadman** [comment]
I'll do it, thanks for the time you've taken on this. ^^

**Comment by geekley** [comment]
> This limitation will likely be overcomed eventually, when Zypak give us a file picker portal [...]

Is this still true?

**Comment by lionirdeadman** [comment]
> > This limitation will likely be overcomed eventually, when Zypak give us a file picker portal [...]
> 
> Is this still true?

I never saw this but no, I'll correct it to say that we're waiting on Electron.

**Pull request merged**
